### PR TITLE
Upgrade dependencies and project

### PR DESCRIPTION
**Pull request by neoziro**
Hello, I am a new collaborator on the project. I use React for three years now, I am the author of [Shipit](https://github.com/shipitjs/shipit) and I give JavaScript trainings in France ([Smooth Code](https://www.smooth-code.com/).

My goal is to maintain this project and release a stable v3.

For my first PR, I did several things:

- Replace mocha by Jest
- Use snapshots for Babel tests
- Use babel-preset-env instead of deprecated babel-preset-es2015
- Upgrade project dependencies (especially source-map)
- Upgrade ESLint and use Prettier
- Target node 6 and node 8 on Travis

@gaearon you look like to be the main contributor on the project, I would be happy to have your quick review before merging it.

Aside, I merged all relevant PR on the project. Next steps will be:

- Take care of all opened issues
- Upgrade documentation
- Release a stable v3

### CODE REVIEW

File: src/AppContainer.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 /* eslint-disable global-require */
 
-'use strict';
```
Comment (gaearon) [inline src/AppContainer.js]: I think we need `use strict` here. If ESLint says we don't, it's wrong.

File: src/AppContainer.prod.js [hunk 2]
```diff
@@ -1,18 +1,17 @@
 /* eslint-disable react/prop-types */
 
-'use strict';
```
Comment (gaearon) [inline src/AppContainer.prod.js]: Same. It's not an ES6 module. It needs `use strict`.

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all",
+  "semi": false
```
Comment (gaearon) [inline .prettierrc]: Please let's keep semis 😛

Comment (neoziro) [reply .prettierrc]: @gaearon OK 😤

Comment (gaearon) [reply .prettierrc]: If you really insist we can remove. You're the maintainer now 😄

Comment (gaearon) [reply .prettierrc]: It's the fate of all my projects now. I pass them on and people remove the semicolons...

File: src/AppContainer.js (hunk 1 above)
Comment (neoziro) [reply src/AppContainer.js]: We transpile it using Babel, so why do we need to add it?

```sh
$ babel src/AppContainer.js                                                                                                                       
'use strict';

/* eslint-disable global-require */

if (!module.hot || process.env.NODE_ENV === 'production') {
  module.exports = require('./AppContainer.prod');
} else {
  module.exports = require('./AppContainer.dev');
}
```

Comment (gaearon) [reply src/AppContainer.js]: Oh okay.

File: .prettierrc (hunk 3 above)
Comment (neoziro) [reply .prettierrc]: It is not a real problem since Prettier does the job for us.

Comment (gaearon) [reply .prettierrc]: Yea I know. Just funny.

**Pull request merged**